src/tools/ScenarioMeasurement/Startup/Startup.cs

```diff
@@ -105,37 +112,105 @@ static void checkArg(string arg, string name)
+                    Executable = appExe,
+                    Arguments = appArgs,
+                    WorkingDirectory = workingDir,
+                    GuiApp = guiApp,
+                    EnvironmentVariables = envVariables
+                };
+            }
+            else
+            {
+                TestProcess = new ManagedProcessHelper(logger)
+                {
+                    ProcessWillExit = processWillExit,
+                    Timeout = timeout,
+                    MeasurementDelay = measurementDelay,
+                    Executable = appExe,
+                    Arguments = appArgs,
+                    WorkingDirectory = workingDir,
+                    GuiApp = guiApp,
+                    EnvironmentVariables = envVariables
+                };
+            }
+
+            IProcessHelper secondTestProcess = metricType == MetricType.InnerLoop || metricType == MetricType.InnerLoopMsBuild ? TestProcess : null;
+
+            //Create wait funcs for steady state and post-compilation
+            Func<Process, string, bool> waitForSteadyState = metricType == MetricType.DotnetWatch ? (Proc, searchString) =>
+            {
+                StringBuilder output = new StringBuilder();
+                DataReceivedEventHandler stdOutProcessor = (s, e) =>
+                {
+                    if (!String.IsNullOrEmpty(e.Data))
+                    {
+                        output.AppendLine(e.Data);
+                        Console.WriteLine(e.Data);
+                    }
+                };
+                DataReceivedEventHandler stdErrProcessor = (s, e) =>
+                {
+                    if (!String.IsNullOrEmpty(e.Data))
+                    {
+                        Console.WriteLine(e.Data);
+                    }
+                };
+                Proc.OutputDataReceived += stdOutProcessor;
+                Proc.ErrorDataReceived += stdErrProcessor;
+                Proc.BeginOutputReadLine();
+                Proc.BeginErrorReadLine();
+                bool isSteadyState = false;
+                int timeoutCount = 0;
+                while (!isSteadyState && timeoutCount < timeout)
+                {
+                    foreach (var line in output.ToString().Split(Environment.NewLine))
+                    {
+                        if (line.Contains(searchString))
+                        {
+                            isSteadyState = true;
+                            break;
+                        }
+                    }
+                    timeoutCount++;
+                    Thread.Sleep(1000);
+                }
+                Proc.CancelErrorRead();
+                Proc.CancelOutputRead();
+                Proc.ErrorDataReceived -= stdErrProcessor;
+                Proc.OutputDataReceived -= stdOutProcessor;
+                return true && timeoutCount < timeout;
+            }
+            : null;
+
+            Func<Process, string, bool> waitForRecompile = waitForSteadyState;
```

Comment:
nit: you don't really need a second name here right?

Reply:
I think I want to keep it because I could imagine wanting to do different things between these two steps. Right now we don't but I would like to keep that option.